File: factorial.rb
#! /usr/bin/ruby
# factorial.rb displays a factorial
#Lorrayya Williams for CS 214 at Calvin College
#
# Input:  A number
# Precondition: The input must be a number
# Output: A factorial of the inputed number
if __FILE__ == $0
  print "Enter the number you wish to see a factorial of "
 number = gets.to_f
 value = number
 original_num = number

 #change to for loop
for i in 2..original_num do
    value -= 1
    number *= value
  end
puts "The factorial of #{original_num} is #{number}."
end

File: factorial.java
/* factorial.java displays the facortial of a number
 *
 * Input: A number that is the type long
 * Output: The Factorial of a Number
 * Written by: Lorrayya Williams for CS 214 at Calvin College
 ********************************************************/
import java.util.Scanner; 		// Get input
class Factorial {

  public static void main(String[] args){
    Scanner keyboard = new Scanner(System.in);
    System.out.println("Enter Number you want factorial of");
    double n;
    n = keyboard.nextLong();

  double answer = 1.0;
  for(int i= 2; i<= n; i++){
    answer = answer * i;
  }
    System.out.println("The facotorial of " + n + " is " + answer);
}
}
